datafusion/core/src/datasource/physical_plan/arrow_file.rs

```diff
@@ -99,7 +99,19 @@ impl FileSource for ArrowSource {
         "arrow"
     }
 
-    impl_schema_adapter_methods!();
+    fn with_schema_adapter_factory(
```

Comment:
instead of the `impl_schema_adapter_methods` macro, I instead replicated the code several places. While there is more duplicated code I think it is clearer because what is happening is now more explicit and doesn't require a macro (even though `impl_schema_adapter_methods` was super well documented)